GVFS/GVFS.Common/ProductUpgrader.cs

```diff
@@ -114,18 +115,15 @@ protected ProductUpgrader()
         /// Deletes any previously downloaded installers in the Upgrader Download directory.
         /// This can include old installers which were downloaded but never installed.
         /// </summary>
-        public static void DeleteAllInstallerDownloads(ITracer tracer = null)
+        public virtual void DeleteAllInstallerDownloads(ITracer tracer)
         {
             try
             {
-                PhysicalFileSystem.RecursiveDelete(ProductUpgraderInfo.GetAssetDownloadsPath());
+                this.fileSystem.RecursiveDelete(ProductUpgraderInfo.GetAssetDownloadsPath());
```

Comment:
Without this change (and the corresponding fallout of having to pass `fileSystem` around) the unit tests were deleting the download path on my dev machine.